src/hex/iter.rs

```diff
@@ -66,3 +66,30 @@ impl<I: Iterator<Item = Hex>> HexIterExt for I {
         self.collect()
     }
 }
+
+/// Private container for a [`Hex`] [`Iterator`] of known size
+#[derive(Debug, Clone)]
+pub struct ExactSizeHexIterator<I> {
```

Comment:
The fact that this is `pub` is rather confusing to me, given that this is described as being private.

Reply:
It is in a private module, innaccessible from library users